### PR TITLE
Fixing `.drone.yml`

### PR DESCRIPTION
**Pull request by dessalines**
This took so many tries, but its ready to go. This includes:

- Combining all the cargo tasks into one step. This seems to be the [recommended way](https://docs.drone.io/pipeline/kubernetes/examples/language/rust/) to handle rust pipelines in drone, and it gets around the cache issue. Maybe not ideal, but also not a big deal, since the most important information is what line / where the build failed, which the log still tells us.
- Ran clippy, and added another lint for builds to pass.
- Fixed the drone docker publish plugin by adding explicit `add_host` IP addresses for several crates.io related domains, since DNS resolution breaks in the drone container, probably because they use an old docker build inside.
- Removed the `cargo install workspaces ... cargo exec` steps, changed them to just do `cargo check --workspace --no-default-features`
- Added a notify on failure block at `ntfy.sh/lemmy_drone_ci`, so I get live notifiations for when a build fails, and a link to the build. You can also add this if you have android, or through the ntfy web client.
- Use simple `alpine:3` image for image prep
- 1.67.0 upgrade
- Postgres 15 upgrade.

Note: I tried many times switching from `clux/muslrust`, to `rust:1.67.0-alpine` (which also is a musl image, but maintained by rust), but unfortunately the binary it built kept segfaulting when running on alpine, with no clear indication of what was wrong.

### CODE REVIEW

**Comment by Nutomic** [comment]
> Combining all the cargo tasks into one step. This seems to be the [recommended way](https://docs.drone.io/pipeline/kubernetes/examples/language/rust/) to handle rust pipelines in drone, and it gets around the cache issue. Maybe not ideal, but also not a big deal, since the most important information is what line / where the build failed, which the log still tells us.

This is not a recommendation but only an example. And its a very bad idea because to figure out which command failed, you need to scroll through thousands of log lines. The current config lets you see that at a glance. So please change it back to separate steps.

File: .drone.yml [hunk 1]
```diff
@@ -40,48 +40,21 @@ steps:
...
-      - cargo test --workspace --no-fail-fast --all-features
-
-  - name: check defaults.hjson updated
-    image: clux/muslrust:1.64.0
-    environment:
-      CARGO_HOME: .cargo
-    commands:
+      - echo "check with different features"
+      - cargo check --workspace --no-default-features
+      - cargo check --workspace --all-features
```
Comment (Nutomic) [inline .drone.yml:47]: Remove these cargo check commands and instead put:
```
cargo check --package lemmy_utils
cargo check --package lemmy_db_schema
cargo check --package lemmy_db_views
cargo check --package lemmy_db_views_actor
cargo check --package lemmy_db_views_moderator
cargo check --package lemmy_api_common
cargo check --package lemmy_api
cargo check --package lemmy_api_crud
cargo check --package lemmy_apub
cargo check --package lemmy_routes
cargo check --package lemmy_server
cargo check --package lemmy_server --all-features
cargo check --package lemmy_server --no-default-features
```

That way we ensure that every package compiles with default features, and that lemmy_server compiles with different feature flags.

File: .drone.yml [hunk 2]
```diff
@@ -10,27 +10,27 @@ steps:
 
   # use minimum supported rust version for most steps
   - name: prepare repo
-    image: clux/muslrust:1.64.0
+    image: alpine:3
```
Comment (Nutomic) [inline .drone.yml:13]: Does this really have any benefit? Anyway it has to download the muslrust image later, so downloading an extra image here will only take extra time and space.

**Comment by dessalines** [comment]
> This is not a recommendation but only an example. And its a very bad idea because to figure out which command failed, you need to scroll through thousands of log lines. The current config lets you see that at a glance. So please change it back to separate steps.

I tried that, and it would time-out my slower builder, because it went over the allotted 2 hours time. It does that because each step has to build from scratch, because drone doesn't cache between steps. You can try to get that working, but it might take another few days, because you have to wait 2 hours for each build to complete.

File: .drone.yml (hunk 1 above)
Comment (dessalines) [reply .drone.yml:47]: Done

File: .drone.yml (hunk 2 above)
Comment (dessalines) [reply .drone.yml:13]: Yep, namely that we'll never have to upgrade it, in the same way we have to change all the rust versions when there's an upgrade. By consolidating all the cargo related steps into one image, we really only have to use that image for one step, and upgrade only that one.

**Comment by Nutomic** [comment]
Did you use `CARGO_HOME: .cargo`? Thing is that drone only saves the repo folder across build steps, but cargo dependencies are stored in the home folder by default. With that env var they are also stored in the repo, so they dont need to be downloaded again. Build artifacts are stored in the repo by default, so no further caching should be necessary.

**Comment by dessalines** [comment]
Yep I did try that, and it would still re-download and compile for each step.

File: .drone.yml (hunk 2 above)
Comment (Nutomic) [reply .drone.yml:13]: Upgrading is a simple replace all so thats not a good argument imo.

**Comment by dessalines** [comment]
You can try separating them again, but again it timed out my builder doing that. You have to scroll to the line anyway to get the specific error when it fails, whether you split these out into separate steps or not. So to me dealing with drone's lack of native rust caching support is a waste of time.

**Pull request merged**